omod/src/main/java/org/openmrs/module/fhir2/web/filter/ForwardingFilter.java

```diff
@@ -33,7 +38,29 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 		String contextPath = ((HttpServletRequest) req).getContextPath();
 		String prefix = contextPath + "/ws/fhir2/";
 		if (requestURI.startsWith(prefix)) {
-			String newURI = requestURI.replace(prefix, "/ms/fhir2Servlet/");
+			String[] requestPath = requestURI.split(prefix, 1);
+			String maybeVersion = requestPath[1].split("/", 1)[0];
+			
+			String replacement;
+			switch (maybeVersion) {
+				case "R2":
+					prefix += "/" + maybeVersion + "/";
+					replacement = "/ms/fhir2Dstu2Servlet";
+					break;
+				case "R3":
+					prefix += "/" + maybeVersion + "/";
+					replacement = "/ms/fhir2Dstu3Servlet";
+					break;
+				case "R4":
+					prefix += "/" + maybeVersion + "/";
+					replacement = "/ms/fhir2Servlet";
+					break;
+				default:
+					replacement = "/ms/fhir2Servlet";
+					break;
```

Comment:
We need to rethink this. We should return an error if the value isn't a known version so that we're using a consistent access URL.